### ISSUE TITLE
Add `concat_ws` scalar function

### ISSUE THREAD

**Issue by seut** [crate/crate#12259]
**Use case**:
Concatenate `TEXT` or to `TEXT` castable type values using a given *separator*, for example concatenate address parts by comma.

```sql
SELECT concat_ws(',', '535 Mission St.', '14th floor', 'San Francisco', 'CA', '94105') AS address;
--> 535 Mission St., 14th floor, San Francisco, CA 94105
```

**Feature description**:
Add support for the `concat_ws ( sep text, val1 "any" [, val2 "any" [, ...] ] )` scalar function which takes a text separator as the first argument and a variable number of any data type arguments whose `TEXT` representation will be concatenated using the given separator. `NULL` values must be ignored.

See also [PostgreSQL's documentation](https://www.postgresql.org/docs/current/functions-string.html):
```
concat_ws ( sep text, val1 "any" [, val2 "any" [, ...] ] ) → text
Concatenates all but the first argument, with  separators. The first argument is used as the separator string, and  should not be NULL. Other NULL arguments are ignored.
concat_ws(',', 'abcde', 2, NULL, 22) → abcde,2,22
```

**Comment by mayraq10** [comment]
@seut  My team at Enfuse would like to take this issue.  Would you assign it to me, please?

**Comment by BaurzhanSakhariev** [comment]
Hi @mayraq10, assigned it to you. Thanks for taking it!

**Comment by mayraq10** [comment]
@BaurzhanSakhariev  We are having issues with the Checkstyle plugin.  We are getting errors when we try to set up the plugin's configurations.  "com.puppycrawl.tools.checkstyle.api.CheckstyleException: unable to parse configuration stream
	at com.puppycrawl.tools.checkstyle.ConfigurationLoader.loadConfiguration(ConfigurationLoader.java:354)"
Any suggestions?
Otherwise, I guess we could redo the pull request and see if we pass.

**Comment by BaurzhanSakhariev** [comment]
Hi @mayraq10, what IDE do you use? If IntelliJ IDEA can run `./gradlew idea` to setup project, including code style.

https://github.com/crate/crate/blob/master/devs/docs/basics.rst#using-an-ide

**Comment by BaurzhanSakhariev** [comment]
Hi @mayraq10, what IDE do you use? If IntelliJ IDEA can run `./gradlew idea` to setup project, including code style.

https://github.com/crate/crate/blob/master/devs/docs/basics.rst#using-an-ide

**Comment by mayraq10** [comment]
Yes, IntelliJ...  I get a BUILD SUCCESSFUL.  Are we supposed to manually add the plugin? Are there some configurations that we are missing?  We are not getting any errors on our IDE, but when we submitted the pull request yesterday, it showed checkstyle errors.

**Comment by BaurzhanSakhariev** [comment]
Do you use `./gradlew checkstyleMain` to check locally?

**Comment by rc-enfuse** [comment]
@BaurzhanSakhariev We didn't know about `./gradlew checkstyleMain`. That command works great however the only place I could find this command referenced is the main.yml file used in your CI/CD process. Thanks for the tip though.

**Comment by BaurzhanSakhariev** [comment]
Closed in #12355. Thanks a lot @mayraq10, @rc-enfuse!
